File: problemsolve/PS_14719.cpp
#include <iostream>
#include <algorithm> 
using namespace std;

int arr[510];

int main(){
	
	int H,W;
	cin >> H >> W;
	
	int cnt = 0;
	
	for(int i=0;i<W;++i){
		cin >> arr[i];  //arr[i]의 값은 쌓인 블록 높이 
	}
	
	for(int i=1;i<W-1;++i){
		
		int currentH = arr[i];
		int leftH = currentH;
		int rightH = currentH;
		
		for(int l=i-1;l>=0;--l){	 // 왼쪽에서 가장 큰 높이 찾기 
			if(arr[l]>currentH){
				leftH = max(leftH, arr[l]);	
			}
		}
		for(int r=i+1;r<W;++r){ 	 // 오른쪽에서 가장 큰 높이 찾기 
			if(arr[r]>currentH){
				rightH = max(rightH, arr[r]);	
			}	
		} 
		
		int set = min(leftH,rightH);
		cnt += (set-currentH);
		
	}
	
	cout << cnt;
	
	return 0;
}

File: problemsolve/solve-to-java/PS_14719.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    static int N,M;
    static int[] heights = new int[510];

    static public int solve(int row,int col,int arr[]){

        int cnt = 0;
        for(int i=1;i<M-1;++i){
            int left = 0, right = 0;
            for(int l=0;l<i;++l){
                left = Math.max(left,arr[l]);
            }
            for(int r=i+1;r<M;++r){
                right = Math.max(right,arr[r]);
            }

            if(left>arr[i] && right>arr[i]) {
                int min_height = Math.min(left, right);
                cnt += min_height - arr[i];
            }
        }
        return cnt;
    }

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());
        st = new StringTokenizer(br.readLine());
        for(int i=0;i<M;++i){
            heights[i] = Integer.parseInt(st.nextToken());
        }
        System.out.println(solve(N,M,heights));
    }
}
